File: 1003. Check If Word Is Valid After Substitutions/Solution.cpp
#include <stack>
#include <string>
using namespace std;

class Solution {
public:
    bool isValid(string s)
    {
        stack<char> stk;
        for (const char& ch : s) {
            if (ch == 'c') {
                if (stk.size() < 2)
                    return false;
                char b = stk.top();
                stk.pop();
                char a = stk.top();
                stk.pop();
                if (a != 'a' || b != 'b')
                    return false;
            } else
                stk.emplace(ch);
        }
        return stk.empty();
    }
};

File: 1003. Check If Word Is Valid After Substitutions/Solution.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for c in s:
            if c == "c":
                if len(stack) < 2 or stack.pop() != "b" or stack.pop() != "a":
                    return False
            else:
                stack.append(c)
        return not stack
